src/Controllers/CreateNewEntryController.php

```diff
@@ -3,41 +3,54 @@
 namespace App\Controllers;
 
 use App\Abstracts\Controller;
-
+use App\Interfaces\BlogModelInstanceInterface;
+use App\Validators\Validators;
 use Psr\Http\Message\ResponseInterface as Response;
 use Psr\Http\Message\ServerRequestInterface as Request;
+use Valitron\Validator;
 
-
-class CreateNewEntryController extends Controller
+class CreateNewEntryController extends Controller implements BlogModelInstanceInterface
 {
-    private $BlogModel;
+    private BlogModelInstanceInterface $blogModel;
 
     /**
-     * CreateNewEntryController constructor.
-     * @param $BlogModel
+     * editEntryController constructor.
+     * @param $blogModel
      */
-    public function __construct($BlogModel)
+    public function __construct(BlogModelInstanceInterface $blogModel)
     {
-        $this->BlogModel = $BlogModel;
+        $this->blogModel = $blogModel;
     }
 
-    public function __invoke(Request $request, Response $response, array $args)
+    public function __invoke(Request $request, Response $response, array $args): Response
     {
-        $ResponseData =
+        $responseData =
             [
                 'success' => false,
                 'message' => '',
                 'data' => []
             ];
 
-        $AddNewEntry = $request->getParsedBody()['NewBlogEntry'] ?? null;
+        $newBlogPost = $request->getParsedBody();
+        $blogPost = new Validator($newBlogPost);

```

Comment:
This variable doesn't accurately represent what this variable contains. i'd say maybe unvalidatedBlogPost, its a bit strange as its creating an object with the blogPost data, ideally you could add this into your ValidateNewPost method so its encapsulated better,